### DISCUSSION TITLE
:question: `firebase_options.dart` の公開について

### DISCUSSION THREAD

**Discussion by chippy-ao** [tatsutakein-jp/asis-app#84] [Q&A] [answered]
現在、flavor対応を行なっています。
手始めにfirebaseの切り替えをしていたのですが、設定ファイルをどうしようかなとなりました。
アップロードする場合、flavor対応の範囲で行いますが、アップロードしない場合、後回しにします。

[flutterfireチームのリポジトリ](https://github.com/firebase/flutterfire/blob/master/packages/firebase_auth/firebase_auth/example/ios/firebase_app_id_file.json)ではプッシュされており、[Firebase 構成ファイルとオブジェクト](https://firebase.google.com/docs/projects/learn-more?hl=ja#config-files-objects)でも公開情報であると記載されています（それと同時に公開リポジトリでは上げない方がいいとも）。

要するにFirestoreなど使用しないにしても、Rulesで全コレクションでFalseを設定したり、AppCheckを入れるなどして、いたずらを防ぐ技術があれば上げてもいいってことかなと。

> [!NOTE]
> Firebase 構成ファイルまたは Firebase 構成オブジェクトの内容は公開情報とみなされます。公開であるとみなされる情報としては、アプリのプラットフォーム固有の ID（Apple バンドル ID または Android パッケージ名）や Firebase プロジェクト固有の値（API キー、プロジェクト ID、Realtime Database URL、Cloud Storage バケット名など）があります。このような点を考慮したうえで、Firebase セキュリティ ルールを使用し、[Realtime Database](https://firebase.google.com/docs/database/security?hl=ja)、[Cloud Firestore](https://firebase.google.com/docs/firestore/security/get-started?hl=ja)、[Cloud Storage](https://firebase.google.com/docs/storage/security?hl=ja) にあるデータとファイルを保護してください。
> 
> オープンソース プロジェクトの場合、一般的に考えて、ソース管理にアプリの Firebase 構成ファイルや Firebase 構成オブジェクトを含めることはおすすめできません。ほとんどの場合、ユーザーは独自の Firebase プロジェクトを作成し、（独自の Firebase 構成ファイルまたは Firebase 構成オブジェクトを使用して）アプリが独自の Firebase リソースを参照するように設定すると想定されるからです。

**Comment by tatsutakein** [comment, marked as answer]
逆コンパイルされたら見られる情報ではあるけれど、構成ファイルは一旦 dev 環境のみの追加として stg, prod は gitignore としましょ！
それなら通常開発する分には困らないと思うので 🙆

AppCheck は https://github.com/tatsutakein-jp/asis-app/issues/22 で Issue を切ってあります 👍

**Comment by chippy-ao** [comment]
@tatsutakein 
了解です。
一旦、自分の組み方でドラフトを提出するので、お待ちください。
あと、Firebaseの設定でiOS側のバンドルにサフィックスが全て入っていなかったので、そこはコンソール側で再生成しておきます。
